cms/djangoapps/contentstore/views/item.py

```diff
@@ -187,9 +201,14 @@ def xblock_handler(request, usage_key_string):
             _delete_item(usage_key, request.user)
             return JsonResponse()
         else:  # Since we have a usage_key, we are updating an existing xblock.
+            discussion_enabled = request.json.get('fields', {}).get('discussion_enabled')
+            xblock = _get_xblock(usage_key, request.user)
+            if isinstance(discussion_enabled, bool) and usage_key.category in ["sequential", "chapter"]:
+                set_discussion_toggle(discussion_enabled, xblock, request.user)
```

Comment:
Do this inside _save_xblock in the `bulk_operations` context manager so that its recorded with the rest of the changes in the transaction.